### PR TITLE
Implement Hilt

### PR DESCRIPTION
**Pull request by naufalprakoso**
## Overview (Required)
In the last few months, Google release Hilt. Hilt is a dependency injection library for Android that reduces the boilerplate of doing manual dependency injection in your project. I think Hilt better than dagger, even though it's built with dagger2.

## Links
- https://developer.android.com/training/dependency-injection/hilt-android
- https://developer.android.com/training/dependency-injection/hilt-jetpack
- https://www.youtube.com/watch?v=B56oV3IHMxg

### CODE REVIEW

**Comment by Pluu** [comment]
@naufalprakoso  Thank you.
Positively consider your wonderful suggestion.

cc @leeyc09 @Jiyoung9310

File: androidapp/dependency.gradle [hunk 1]
```diff
@@ -7,14 +7,15 @@ ext.deps = [:]
...
-versions.dagger = "2.25.4"
 versions.firebase = "17.2.2"
 versions.firestore = "21.3.1"
 versions.lifecycle = "2.2.0"
 versions.coroutine = "1.3.3"
 versions.navigation = "2.2.0"
 versions.timber = "4.7.1"
 versions.browser = "1.2.0"
+versions.hilt = "2.28-alpha"
+versions.hilt_view_model = "1.0.0-alpha01"
```
Comment (Pluu) [inline androidapp/dependency.gradle:18]: @naufalprakoso Newest version is "1.0.0-alpha02"
https://developer.android.com/jetpack/androidx/releases/hilt#version_100_2

File: androidapp/dependency.gradle [hunk 2]
```diff
@@ -7,14 +7,15 @@ ext.deps = [:]
...
 versions.kotlin = "1.3.61"
-versions.dagger = "2.25.4"
 versions.firebase = "17.2.2"
 versions.firestore = "21.3.1"
 versions.lifecycle = "2.2.0"
 versions.coroutine = "1.3.3"
 versions.navigation = "2.2.0"
 versions.timber = "4.7.1"
 versions.browser = "1.2.0"
+versions.hilt = "2.28-alpha"
```
Comment (Pluu) [inline androidapp/dependency.gradle:17]: @naufalprakoso Newest version is "2.28.3-alpha"
https://mvnrepository.com/artifact/com.google.dagger/hilt-android-compiler

**Comment by naufalprakoso** [comment]
Wow, thanks for your appreciation of my work. I am loved to contribute. I think you have good English 😁
By the way, thank you for review my PR. And I already rebased the PR, please check my PR again. Thank you!

**Comment by Jiyoung9310** [comment]
다른 코드가 머지되면서 충돌난 부분 해결하여 master에 반영하였습니다

**Pull request merged**